### DISCUSSION TITLE
watch mode and backups_to_keep_remote

### DISCUSSION THREAD

**Discussion by romaka** [Altinity/clickhouse-backup#999] [Q&A] [answered]
Hey guys,

does the clihkouse-backup being started in `watch` mode respect `backups_to_keep_remote` configuration directive? 

I mean, should we expect  the utility to remove older backups and keep overall backups count equal to `backups_to_keep_remote` value?

For instance, to keep one full and seven incremental backups for two weeks, we set `backups_to_keep_remote` to 16 (2 full + 14 incremental) and start `watch` with `full_interval=168h` / `watch_interval=24h`. But after two weeks of watch process running, the number of backups just keeps increasing so no retention takes place, we suppose.

Do we miss something conceptual and expect something that is not supposed to work that way?

Thanks.

**Comment by Slach** [comment, marked as answer]
Logic for backup retention is simple

during deletion, `old` backup will delete only if no references to him in `live` backups, to avoid break reference in incremental backup sequence
actually you have 2 full + 12 incremental 


try to

```yaml
general:
 backups_to_keep_remote: 8
 backups_to_keep_local: -1
```

In this case, you will always have 2 full backups 1full +6 incremental in previous week and 1 full + from 1 to 6 of incremental backups in last week depends on current weekday
after third full backup creation, 1+6 incremental will drop

**Reply by romaka** [reply]
Thank you, mate. I've tried these values as you suggested and everything looks good now.
